app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    if params[:search_app].present?
+      @pets = Pet.search(params[:search_app])
+    end
+
+    if params[:pet_id].present?
```

Comment:
This controller action is doing a bit too much. `show` actions are for reading data so it shoudn't handle creating the join record between a pet and application. Instead, you can have that button to Adopt a Pet route to a joins controller, called ApplicationPetsController or PetApplicationsController. You could route it to a `create` action there to indicate the creation of a joins resource.

Reply:
Then, you can route the app back to the show page